uPortal-portlets/src/main/java/org/apereo/portal/portlets/portletadmin/PortletAdministrationHelper.java

```diff
@@ -115,10 +115,14 @@
 
     private static final String PORTLET_FNAME_FRAGMENT_ADMIN_PORTLET = "fragment-admin";
 
-    public static final String[] PORTLET_SUBSCRIBE_ACTIVITIES = {
+    public static final String[] PORTLET_PRINCIPAL_SUBSCRIBE_ACTIVITIES = {
         IPermission.PORTLET_SUBSCRIBER_ACTIVITY, IPermission.PORTLET_BROWSE_ACTIVITY
     };
 
+    public static final String[] PORTLET_PRINCIPAL_PUBLISH_ACTIVITIES = {
+        IPermission.PORTLET_MODE_CONFIG
+    };
+
```

Comment:
I believe you can tackle this change in a simpler way...

It appears that the `PORTLET_SUBSCRIBE_ACTIVITIES` (before you renamed it) array is only used in this class.  It appears, furthermore, that the role of the array is only about what things to add to a web form and process.

The name is misleading.  It is in no way important that the members of the array are all `SUBSCRIBE` permissions -- it's merely a coincidence.

I would...

  - Make the variable `private`
  - Change the name to something like `PORTLET_PERMISSIONS_ON_FORM`
  - Add your new one to the original array.

Reply:
In case you haven't already figured it out, I was full of it in this remark.  What you have is pretty good.

Reply:
It was a bit brittle.  I'm currently revising it to use an enum with the plan to have both a single group of possible portlet permissions and the ability to differentiate the owners.